samples/cpp/benchmark_app/main.cpp

```diff
@@ -747,88 +747,91 @@ int main(int argc, char* argv[]) {
+                            }
                         }
                     }
+                    // Explicitly set inputs layout.
+                    if (!name.empty() && !app_inputs_info[0].at(name).layout.empty()) {
+                        in.model().set_layout(app_inputs_info[0].at(name).layout);
+                    }
                 }
-                // Explicitly set inputs layout.
-                if (!name.empty() && !app_inputs_info[0].at(name).layout.empty()) {
-                    in.model().set_layout(app_inputs_info[0].at(name).layout);
-                }
-            }
 
-            fuse_mean_scale(preproc, app_inputs_info.at(0));
+                fuse_mean_scale(preproc, app_inputs_info.at(0));
+
+                const auto& outs = model->outputs();
+                for (size_t i = 0; i < outs.size(); i++) {
+                    const auto& item = outs[i];
+                    auto iop_precision = ov::element::dynamic;
+                    std::string name;
+                    try {
+                        // Some tensors might have no names, get_any_name will throw exception in that case.
+                        // -iop option will not work for those tensors.
+                        name = item.get_any_name();
+                    } catch (...) {
+                    }
+                    const auto& prc = user_precisions_map.find(name);
+                    if (prc != user_precisions_map.end()) {
+                        iop_precision = getPrecision2(prc->second);
+                    }
 
-            const auto& outs = model->outputs();
-            for (size_t i = 0; i < outs.size(); i++) {
-                const auto& item = outs[i];
-                auto iop_precision = ov::element::dynamic;
-                std::string name;
-                try {
-                    // Some tensors might have no names, get_any_name will throw exception in that case.
-                    // -iop option will not work for those tensors.
-                    name = item.get_any_name();
-                } catch (...) {
-                }
-                const auto& prc = user_precisions_map.find(name);
-                if (prc != user_precisions_map.end()) {
-                    iop_precision = getPrecision2(prc->second);
+                    if (iop_precision != ov::element::dynamic) {
+                        preproc.output(i).tensor().set_element_type(iop_precision);
+                    } else if (output_precision != ov::element::dynamic) {
+                        preproc.output(i).tensor().set_element_type(output_precision);
+                    }
                 }
 
-                if (iop_precision != ov::element::dynamic) {
-                    preproc.output(i).tensor().set_element_type(iop_precision);
-                } else if (output_precision != ov::element::dynamic) {
-                    preproc.output(i).tensor().set_element_type(output_precision);
-                }
+                model = preproc.build();
```

Comment:
Pavel mentioned that `preproc` holds the model. @praasz, maybe `preproc`'s constructor could consume the model in constructor and move away in `build()`?

Reply:
Not sure how to define that for Python